mbedtls/src/cipher/raw/serde.rs

```diff
@@ -87,6 +90,35 @@ impl<Op: Operation> Serialize for Cipher<Op, Traditional, CipherData> {
                 | (CIPHER_ID_3DES, MODE_CFB) => AlgorithmContext::Des3(Bytes(
                     *(cipher_context.cipher_ctx as *const des3_context),
                 )),
+                (CIPHER_ID_AES, MODE_GCM) => {
+                    let gcm_context = *(cipher_context.cipher_ctx as *const gcm_context);
+
+                    let mut inner_ctx = gcm_context.cipher_ctx;
+
+                    let inner_ctx_cipher_id = (*(*inner_ctx.cipher_info).base).cipher;
+                    let inner_ctx_cipher_mode = (*inner_ctx.cipher_info).mode;
+                    let inner_ctx_key_bit_len = (*inner_ctx.cipher_info).key_bitlen;
+
+                    let mut inner_aes_context = *(inner_ctx.cipher_ctx as *const aes_context);
+                    inner_aes_context.rk = ::core::ptr::null_mut();
+
+                    inner_ctx.cipher_ctx = ::core::ptr::null_mut();
+                    inner_ctx.add_padding = None;
+                    inner_ctx.get_padding = None;
+
+                    let inner_cipher_ctx = SavedRawCipher {
+                        cipher_id: inner_ctx_cipher_id,
+                        cipher_mode: inner_ctx_cipher_mode,
+                        key_bit_len: inner_ctx_key_bit_len,
+                        context: Bytes(inner_ctx),
+                        algorithm_ctx: AlgorithmContext::Aes(Bytes(inner_aes_context)),
+                    };
+
+                    AlgorithmContext::Gcm {
+                        context: Bytes(*(cipher_context.cipher_ctx as *const gcm_context)),
+                        inner_cipher_ctx: Box::new(inner_cipher_ctx)
+                    }
+                },
```

Comment:
This seems like a mine field and I don't feel qualified to review it. @jethrogb please review this part as well as the deserialize part.